backends/platform/libretro/include/config.h

```diff
@@ -0,0 +1,60 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef CONFIG_H
```

Comment:
I'd suggest naming it `LIBRETRO_CONFIG_H`

Reply:
Actually this file intention is to replace the config.h auto-generated during ScummVM normal builds by configure script, should it remain as it is I guess?

Reply:
Another option could be to not define `HAVE_CONFIG_H` but `NONSTANDARD_PORT` and have a portdefs.h file with all system headers and such defines.
I am not sure which one is better as you don't use the `configure` script.

Reply:
`portdefs.h` is already used as well, but if we move definitions from `config.h` there and not define `HAVE_CONFIG_H`, we get trapped into the "endianess unknown" error [here](https://github.com/scummvm/scummvm/blob/ab666139e2904a57c1db30d8f020137acbd1a5a7/common/scummsys.h#L263-L302).
I think the correct condition in the above would be `#if !defined(HAVE_CONFIG_H) && !defined(SCUMM_LITTLE_ENDIAN) && !defined(SCUMM_BIG_ENDIAN)`. In that case with all definitions in `portdefs.h` and `HAVE_CONFIG_H` undefined it builds successfully.
If that is the way to go, I'm ok with that and I can implement in this PR.

Reply:
Indeed...
I don't know what is best here. Patch this define or stick with `config.h`.
The latter is the simpler as it's already done.